src/test/java/com/microsoft/sqlserver/jdbc/ActivityIDTest.java

```diff
@@ -0,0 +1,167 @@
+                });
+                try {
+                    latchPool.await();
+                } catch (InterruptedException e) {
+                    fail(e.toString());
+                } finally {
+                    if (null != ds) {
+                        es.shutdown();
+                        ds.close();
+                    }
+                }
+                latchPoolOuterThread.countDown();
+            }
+        });
+        t.run();
+        latchPoolOuterThread.await();
+
+        try {
+            try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
+                stmt.execute("SELECT @@VERSION AS 'SQL Server Version'");
+            }
+        } catch (SQLException e) {
+            fail(e.toString());
+        }
+
+        try {
+            assertEquals(0, ActivityCorrelator.getActivityIdTlsMap().size());
+        } catch (ComparisonFailure e) {
+            Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
+            System.out.println("List of threads alive:");
+            for (Thread thread : threadSet) {
+                System.out.println(thread.toString());
+            }
+            System.out.println("List of entries in the ActivityID map:");
+            System.out.println(ActivityCorrelator.getActivityIdTlsMap().toString());
+            fail(e.toString());
+        }
+    }
+
+    @Test
+    public void testActivityIDPooledConnection() throws Exception {
+        int poolsize = 10;
+        int numPooledExecution = 200;
+
+        PooledConnection pooledCon = ((SQLServerConnectionPoolDataSource) dsPool).getPooledConnection();
+        ExecutorService es = Executors.newFixedThreadPool(poolsize);
+        try {
+            CountDownLatch latchPool = new CountDownLatch(numPooledExecution);
+            es.execute(() -> {
+                IntStream.range(0, numPooledExecution).forEach(i -> {
+                    try (Connection c = pooledCon.getConnection(); Statement s = c.createStatement(); ResultSet rs = s
+                            .executeQuery("SELECT @@VERSION AS 'SQL Server Version'")) {} catch (SQLException e) {
+                        fail(e.toString());
+                    }
+                    latchPool.countDown();
+                });
+            });
+            latchPool.await();
+        } finally {
+            es.shutdown();
+            pooledCon.close();
+        }
+        assertEquals(0, ActivityCorrelator.getActivityIdTlsMap().size());
+    }
+
+    @AfterAll
+    public static void teardown() throws Exception {
+        String activityIDTraceOff = Util.ActivityIdTraceProperty + "=off";
+        try (InputStream is = new ByteArrayInputStream(activityIDTraceOff.getBytes());) {
+            LogManager lm = LogManager.getLogManager();
+            lm.readConfiguration(is);
+        }
+    }
+
+    @BeforeAll
+    public static void testSetup() throws Exception {
+        String activityIDTraceOn = Util.ActivityIdTraceProperty + "=on";

```

Comment:
Could you please also rename Util > `ActivityIdTraceProperty` to `activityIdTraceProperty`

Reply:
done